File: python/word-count/word_count.py
import string, re
def count_words(sentence):
    sentence = [x.strip(string.punctuation) for x in [word for word in re.split(r'[_,\s]+', sentence.casefold()) if len(word.strip(string.punctuation + string.whitespace)) >= 1]]
    return {k : sentence.count(k) for k in list(dict.fromkeys(sentence))}

File: elixir/word-count/lib/word_count.ex
defmodule WordCount do
  @splitters String.graphemes(" _!&@:;'\"[]()_=+,$&%^")

  @doc """
  Count the number of words in the sentence.

  Words are compared case-insensitively.
  """
  @spec count(String.t()) :: map
  def count(sentence) do
    sentence
    |> String.downcase()
    |> String.split(@splitters)
    |> Enum.filter(&(String.length(&1) > 0))
    |> Enum.reduce(%{}, fn word, count -> Map.update(count, word, 1, &(&1 + 1)) end)
  end
end
